src/mesh_analysis.rs

```diff
@@ -144,6 +147,78 @@ pub fn triangulated_mesh_genus(vertex_count: usize, edge_count: usize, face_coun
     1 - (cast_i32(vertex_count) - cast_i32(edge_count) + cast_i32(face_count)) / 2
 }
 
+/// Checks if two geometries are visually identical.
+///
+/// The two geometries can
+/// possibly contain the same vertices and normals bound to the same faces, but all
+/// of these can be in different order, including shifts in face winding. This
+/// function takes all these cases in consideration. Therefore it is slow.
+#[allow(dead_code)]
+pub fn are_visually_identical(g_1: &Geometry, g_2: &Geometry) -> bool {
+    struct UnpackedFace {
+        vertices: (Point3<f32>, Point3<f32>, Point3<f32>),
+        normals: (Vector3<f32>, Vector3<f32>, Vector3<f32>),
+    }
+
+    impl PartialEq for UnpackedFace {
+        fn eq(&self, other: &Self) -> bool {
+            (self.vertices.0 == other.vertices.0
+                && self.vertices.1 == other.vertices.1
+                && self.vertices.2 == other.vertices.2
+                && self.normals.0 == other.normals.0
+                && self.normals.1 == other.normals.1
+                && self.normals.2 == other.normals.2)
+                || (self.vertices.0 == other.vertices.1
+                    && self.vertices.1 == other.vertices.2
+                    && self.vertices.2 == other.vertices.0
+                    && self.normals.0 == other.normals.1
+                    && self.normals.1 == other.normals.2
+                    && self.normals.2 == other.normals.0)
+                || (self.vertices.0 == other.vertices.2
+                    && self.vertices.1 == other.vertices.0
+                    && self.vertices.2 == other.vertices.1
+                    && self.normals.0 == other.normals.2
+                    && self.normals.1 == other.normals.0
+                    && self.normals.2 == other.normals.1)
+        }
+    }
+
+    let mut unpacked_faces_1 = g_1.faces().iter().map(|face| match face {
+        Face::Triangle(f) => Some(UnpackedFace {
```

Comment:
I think the `Some` wrapping is unnecessary, same below.

Reply:
mind blown!